training-app/app/controllers/api/v1/books_controller.rb

```diff
@@ -0,0 +1,18 @@
+module Api
+  module V1
+    class BooksController < ApplicationController
+      before_action :authenticate_user!
```

Comment:
I would extract this to an ApiController or something that inherits from ApplicationController since you will need this in almost every controller from this API